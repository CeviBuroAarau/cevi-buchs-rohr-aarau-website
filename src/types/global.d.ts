declare let global: any; // eslint-disable-line @typescript-eslint/no-explicit-any
