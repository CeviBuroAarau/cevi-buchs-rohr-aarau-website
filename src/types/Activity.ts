export interface Activity {
  title: string;
  url: string;
  thumb: string;
}

export interface CockpitActivities {
  entries: ActivityRaw[];
}

export interface ActivityRaw {
  title: string;
  image: {
    path: string;
  };
  thumb: {
    path: string;
  };
}
