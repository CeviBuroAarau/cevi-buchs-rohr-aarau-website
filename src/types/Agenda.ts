export interface Agenda {
  title: string;
  text: string;
  date: Date;
}

export interface CockpitAgenda {
  entries: Agenda[];
}

export interface CockpitScope {
  display: string;
}

export interface CockpitEventInfo {
  scope: CockpitScope;
  text: string;
  date: Date;
}

export interface EventInfo {
  scope: string;
  text: string;
  date: Date;
}

export interface CockpitEventInfos {
  entries: CockpitEventInfo[];
}
