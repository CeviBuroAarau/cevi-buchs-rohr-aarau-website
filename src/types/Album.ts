export interface Album {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: AlbumImage[];
  previewImage: AlbumPreviewImage;
}

export interface AlbumImage {
  title: string;
  url: string;
}

export interface AlbumPreviewImage {
  url: string;
}

export interface CockpitAlbum {
  entries: CockpitAlbumEntry[];
}

export interface CockpitAlbumEntry {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: CockpitAlbumImage[];
  previewImage: CockpitAlbumPreviewImage;
}

export interface CockpitAlbumImage {
  meta: {
    title: string;
    asset: string;
  };
  path: string;
}

export interface CockpitAlbumPreviewImage {
  path: string;
}
