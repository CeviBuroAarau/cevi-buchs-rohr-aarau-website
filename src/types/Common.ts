export enum FormState {
  NotDisplayed,
  Displayed,
}
