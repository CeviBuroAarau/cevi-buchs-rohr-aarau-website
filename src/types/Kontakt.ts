export interface KontaktFormRequest {
  form: {
    name: string;
    email: string;
    message: string;
  };
}

export interface KontaktFormResponse {
  error: string;
  data: {
    name: string;
    email: string;
    message: string;
  };
}
