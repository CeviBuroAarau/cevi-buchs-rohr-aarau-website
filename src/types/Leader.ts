export interface Leader {
  name: string;
  isactive: boolean;
  scoutname: string;
  function: string[];
  group: string;
  birthyear: string;
  place: string;
  profession: string;
  recreation: string;
  inScoutsSince: string;
  inScoutsBecause: string;
  bestExperiences: string;
  file: string;
}

export interface LeaderRaw {
  name: string;
  isactive: boolean;
  scoutname: string;
  function: string[];
  group: {
    display: string;
  };
  birthyear: string;
  place: string;
  profession: string;
  recreation: string;
  inScoutsSince: string;
  inScoutsBecause: string;
  bestExperiences: string;
  image: {
    path: string;
  };
}

export interface CockpitLeaders {
  entries: LeaderRaw[];
}
