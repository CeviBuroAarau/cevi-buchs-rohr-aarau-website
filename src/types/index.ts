export * from "./Activity";
export * from "./Agenda";
export * from "./Album";
export * from "./Media";
