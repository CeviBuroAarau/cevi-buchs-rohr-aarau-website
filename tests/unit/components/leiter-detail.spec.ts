/* eslint-disable @typescript-eslint/no-explicit-any */
import { shallowMount, Wrapper } from "@vue/test-utils";
import LeiterDetail from "@/components/leiter-detail.vue";
import { Leader } from "@/types";

describe("LeiterDetail Component", () => {
  const leiter: Leader = {
    name: "Leah Dellenbach",
    isactive: true,
    scoutname: "Luna",
    function: ["Gruppenleiterin", "Abteilungsleiterin"],
    group: "Spartacus",
    birthyear: "1998",
    place: "Aarau",
    profession: "Neue Kantonsschule Aarau",
    recreation: "Lesen, Jungschi, mit Freunden treffen",
    inScoutsSince: "ungef\\u00e4hr 2006",
    inScoutsBecause: "es lustig ist mit Gleichgesinnten etwas zu machen",
    bestExperiences: "Pfila 2007, Leiteranl\\u00e4sse",
    file:
      process.env.VUE_APP_COCKPIT_FILES +
      "\\/storage\\/uploads\\/2021\\/02\\/01\\/2017-12-Luna.jpg_uid_6018765b6b961.webp",
  };

  test("render leiter-detail with leiter", () => {
    const wrapper = shallowMount(LeiterDetail, {
      stubs: [],
      propsData: {
        leiter: leiter,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("open Form", async () => {
    const wrapper: Wrapper<LeiterDetail & { [key: string]: any }> =
      shallowMount(LeiterDetail, {
        stubs: [],
        propsData: {
          leiter: leiter,
        },
      });

    await wrapper.vm.open();

    const container = wrapper.find("div.is-active");
    expect(container.exists()).toBe(true);

    const onLeiterOpenedEmitted = wrapper.emitted().onLeiterOpened;
    if (onLeiterOpenedEmitted === undefined) {
      fail("onLeiterOpened mustn't be undefined");
    } else {
      expect(onLeiterOpenedEmitted.length).toBe(1);
    }
  });

  test("close Form", async () => {
    const wrapper: Wrapper<LeiterDetail & { [key: string]: any }> =
      shallowMount(LeiterDetail, {
        stubs: [],
        propsData: {
          leiter: leiter,
        },
      });

    await wrapper.vm.close();

    const container = wrapper.find("div.is-active");
    expect(container.exists()).toBe(false);

    const onLeiterClosedEmitted = wrapper.emitted().onLeiterClosed;
    if (onLeiterClosedEmitted === undefined) {
      fail("onLeiterClosed mustn't be undefined");
    } else {
      expect(onLeiterClosedEmitted.length).toBe(1);
    }
  });
});
