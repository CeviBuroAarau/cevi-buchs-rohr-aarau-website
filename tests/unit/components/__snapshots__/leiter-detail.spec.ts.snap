// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeiterDetail Component render leiter-detail with leiter 1`] = `
<div
  class="modal"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-card"
  >
    <header
      class="modal-card-head"
    >
      <p
        class="modal-card-title"
      >
        
        Leah Dellenbach v/o Luna
      
      </p>
       
      <button
        aria-label="close"
        class="delete"
      />
    </header>
     
    <section
      class="modal-card-body"
    >
      <img
        alt="Luna"
        src="https://cockpit.cevi-buro-aarau.ch/\\\\/storage\\\\/uploads\\\\/2021\\\\/02\\\\/01\\\\/2017-12-Luna.jpg_uid_6018765b6b961.webp"
      />
       
      <table
        class="table is-bordered is-striped is-narrow is-hoverable is-fullwidth"
        role="presentation"
      >
        <tr>
          <td>
            Funktion
          </td>
           
          <td>
            Gruppenleiterin, Abteilungsleiterin
          </td>
        </tr>
         
        <tr>
          <td>
            Gruppe
          </td>
           
          <td>
            Spartacus
          </td>
        </tr>
         
        <tr>
          <td>
            Jahrgang
          </td>
           
          <td>
            1998
          </td>
        </tr>
         
        <tr>
          <td>
            Wohnort
          </td>
           
          <td>
            Aarau
          </td>
        </tr>
         
        <tr>
          <td>
            Beruf/Schule
          </td>
           
          <td>
            Neue Kantonsschule Aarau
          </td>
        </tr>
         
        <tr>
          <td>
            Hobbys
          </td>
           
          <td>
            Lesen, Jungschi, mit Freunden treffen
          </td>
        </tr>
         
        <tr>
          <td>
            In der Jungschi seit
          </td>
           
          <td>
            ungef\\u00e4hr 2006
          </td>
        </tr>
         
        <tr>
          <td>
            In der Jungschi weil
          </td>
           
          <td>
            es lustig ist mit Gleichgesinnten etwas zu machen
          </td>
        </tr>
         
        <tr>
          <td>
            Tollste Cevi Erlebnisse
          </td>
           
          <td>
            Pfila 2007, Leiteranl\\u00e4sse
          </td>
        </tr>
      </table>
    </section>
     
    <footer
      class="modal-card-foot"
    >
      <button
        class="button"
      >
        Schliessen
      </button>
    </footer>
  </div>
</div>
`;
